### PR TITLE
Refactor object type names to match attribute enum names

### PR DESCRIPTION
**Pull request by kcudnik**
For example: SAI_OBJECT_TYPE_HOSTIF_TRAP_GROUP
will match enum sai_hostif_trap_group_attr_t

proposed rule: object name "foo" should translate to
obejct type: SAI_OBJECT_TYPE_FOO
and attribute enum: sai_foo_attr_t

There is a little inconsistency with fdb/route/neighbor (non object id) entries name

```
SAI_OBJECT_TYPE_FDB_ENTRY -> sai_fdb_entry_attr_t  (obj = sai_fdb_entry_t)
SAI_OBJECT_TYPE_ROUTE -> sai_route_attr_t (obj = sai_unicast_route_entry_t)
SAI_OBJECT_TYPE_NEIGHBOR -> sai_neighbor_attr_t (obj = sai_neighbor_entry_t)
```

Would be nice to have names consistent, so here is proposal:
eather change FDB_ENTRY to FDB and sai_fdb_entry_attr_t   to sai_fdb_attr_t
or convert ROUTE to UNICAST_ROUTE_ENTRY and NEIGHBOR to NEIGHBOR_ENTRY

### CODE REVIEW

**Comment by kcudnik** [comment]
Proposed unification for non object id's:

```
SAI_OBJECT_TYPE_FDB_ENTRY -> sai_fdb_entry_attr_t  (obj = sai_fdb_entry_t)
SAI_OBJECT_TYPE_UNICAST_ROUTE_ENTRY -> sai_unicast_route_entry_attr_t (obj = sai_unicast_route_entry_t)
SAI_OBJECT_TYPE_NEIGHBOR_ENTRY -> sai_neighbor_entry_attr_t (obj = sai_neighbor_entry_t)
```

Or this: 
```
SAI_OBJECT_TYPE_FDB_ENTRY -> sai_fdb_entry_attr_t  (obj = sai_fdb_entry_t)
SAI_OBJECT_TYPE_ROUTE_ENTRY -> sai_route_entry_attr_t (obj = sai_route_entry_t)
SAI_OBJECT_TYPE_NEIGHBOR_ENTRY -> sai_neighbor_entry_attr_t (obj = sai_neighbor_entry_t)
```

Do route needs to be called unicast ?

File: inc/saiqueue.h [hunk 1]
```diff
@@ -81,111 +79,108 @@ typedef enum _sai_queue_attr_t
...
-
 } sai_queue_attr_t;
 
-
-
 /**
  * @brief Enum defining statistics for Queue.
  */
 
 typedef enum _sai_queue_stat_counter_t
```
Comment (lguohan) [inline inc/saiqueue.h]: change to sai_queue_stat_counter to sai_queue_stat_, so that we do not need to change the enum

Comment (kcudnik) [inline inc/saiqueue.h]: Any change is fine by me as long as we keep it consistent. Please take time to review this change and let me know what else should we keep/change

File: inc/saivlan.h [hunk 2]
```diff
@@ -129,30 +126,27 @@ typedef enum _sai_vlan_member_attr_t {
...
 
-
 } sai_vlan_member_attr_t;
 
-
-
 /**
  *  @brief VLAN counter IDs in sai_get_vlan_stat_counters() call
  */
 typedef enum _sai_vlan_stat_counter_t
```
Comment (lguohan) [inline inc/saivlan.h]: >_counter [](start = 27, length = 8)

remove _counter_ here

File: inc/saiqueue.h (hunk 1 above)
Comment (lguohan) [inline inc/saiqueue.h]: remove the _counter_ from _sai_queue_stat_counter_t

---
In reply to: [78670968](https://github.com/opencomputeproject/SAI/pull/208#discussion_r78670968) [](ancestors = 78670968)

File: inc/saiport.h [hunk 3]
```diff
@@ -541,167 +539,165 @@ typedef enum _sai_port_attr_t
...
     /* --*/
     SAI_PORT_ATTR_CUSTOM_RANGE_END
 
-
 } sai_port_attr_t;
 
 /**
  *  @brief Port counter IDs in sai_get_port_stat_counters() call
  */
 typedef enum _sai_port_stat_counter_t
```
Comment (lguohan) [inline inc/saiport.h]: >_counter_ [](start = 27, length = 9)

remove _counter_

File: inc/saipolicer.h [hunk 4]
```diff
@@ -145,48 +141,41 @@ typedef enum _sai_policer_attr_t
     /* --*/
     SAI_POLICER_ATTR_CUSTOM_RANGE_END
 
-
 } sai_policer_attr_t;
 
-
-
 /** @brief Enum defining policer statistics */
 typedef enum _sai_policer_stat_counter_t
```
Comment (lguohan) [inline inc/saipolicer.h]: >_counter_ [](start = 30, length = 9)

remove _counter_

File: inc/saibuffer.h [hunk 5]
```diff
@@ -54,32 +54,32 @@ typedef enum _sai_ingress_priority_group_attr_t
 typedef enum _sai_ingress_priority_group_stat_counter_t
```
Comment (lguohan) [inline inc/saibuffer.h]: >_counter_ [](start = 45, length = 9)

remove _counter

**Comment by kcudnik** [comment]
Counter prefix was removed from all stats enum

**Comment by lguohan** [comment]
:shipit:

**Pull request merged**